bundle/config/mutator/resourcemutator/resource_mutator.go

```diff
@@ -35,16 +35,22 @@ func applyInitializeMutators(ctx context.Context, b *bundle.Bundle) diag.Diagnos
 		// OR corresponding fields on ForEachTask if that is present
 		// Overrides job compute settings with a specified cluster ID for development or testing
 		OverrideCompute(),
+	)
 
-		// Reads (dynamic): resources.dashboards.* (checks for existing parent_path and embed_credentials)
-		// Updates (dynamic): resources.dashboards.*.parent_path (sets to workspace.resource_path if not set)
-		// Updates (dynamic): resources.dashboards.*.embed_credentials (sets to false if not set)
-		ConfigureDashboardDefaults(),
+	defaults := []struct {
+		pattern string
+		value   any
+	}{
+		{"resources.dashboards.*.parent_path", b.Config.Workspace.ResourcePath},
+		{"resources.dashboards.*.embed_credentials", false},
+		{"resources.volumes.*.volume_type", "MANAGED"},
+	}
 
-		// Reads (dynamic): resources.volumes.* (checks for existing volume_type)
-		// Updates (dynamic): resources.volumes.*.volume_type (sets to "MANAGED" if not set)
-		ConfigureVolumeDefaults(),
+	for _, defaultDef := range defaults {
+		diags = diags.Extend(bundle.SetDefault(ctx, b, defaultDef.pattern, defaultDef.value))
+	}
 
+	diags = diags.Extend(bundle.ApplySeq(ctx, b,
```

Comment:
Behavior change -- `ApplySeq` breaks on errors, so breaking up the sequence can mean applying the sequence below regardless of errors in the sequence above. A condition on `diags.Error` can be used to return early.

Reply:
good point, fixed in 1f4a9e7f89b7cdb610d16934db2b29f2223465a5